src/gui/PropertyModel.cpp

```diff
@@ -0,0 +1,64 @@
+#include <PropertyModel.h>
+#include <iostream>
+PropertyModel::PropertyModel(QObject* parent) : QAbstractTableModel(parent)
+{
+
+}
+
+PropertyModel::~PropertyModel()
+{
+
+}
+
+int PropertyModel::rowCount(const QModelIndex& ) const
+{
+    return properties.size();
+}
+
+int PropertyModel::columnCount(const QModelIndex& ) const
+{
+    return 2;
+}
+
+QVariant PropertyModel::data(const QModelIndex& index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        if (index.column() == 0)
+            return properties[index.row()];
+        else if (index.column() == 1)
+            return propertySource->property(properties[index.row()].toStdString().c_str());
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags PropertyModel::flags(const QModelIndex& index) const
+{
+    if (index.column() == 1)
+        return QAbstractTableModel::flags(index) | Qt::ItemIsEditable;
+
+    return QAbstractTableModel::flags(index);
```

Comment:
What about:

```c++
return (1 == index.column())
        ? QAbstractTableModel::flags(index) | Qt::ItemIsEditable
        : QAbstractTableModel::flags(index);
```

Reply:
Your version seems less readable to me, but that's just a matter of preference. If this is the preferred style for CDS I will change it

Reply:
I just wanted to simplify control flow. Return once, instead of returning multiple times.

Reply:
It's a matter of preference, so it may remain as it is. Both are fine and I can think of another (with "ret" variable).

Reply:
Oh no, don't go that way... [NR.2](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nr2-dont-have-only-a-single-return-statement-in-a-function).